### PR TITLE
[1.x] Switch to `defineModel` in Vue stubs

### PR DESCRIPTION
**Pull request by osbre**
https://blog.vuejs.org/posts/vue-3-3

### CODE REVIEW

**Comment by jessarcher** [comment]
I'm excited about `defineModel` but I think we should keep this as a draft until the feature is no longer marked experimental in Vue.

**Comment by selenofobico** [comment]
### **_How can I use the [ modal ] component in Breeze, any example of how to use it?_**

**Comment by osbre** [comment]
Stable in 3.4: https://blog.vuejs.org/posts/vue-3-4
Will update the PR soon.

File: stubs/inertia-vue-ts/resources/js/Components/TextInput.vue [hunk 1]
```diff
@@ -1,11 +1,7 @@
 <script setup lang="ts">
 import { onMounted, ref } from 'vue';
 
-defineProps<{
-    modelValue: string;
-}>();
-
-defineEmits(['update:modelValue']);
+const modelValue = defineModel();
```
Comment (Jubeki) [inline stubs/inertia-vue-ts/resources/js/Components/TextInput.vue:4]:
```suggestion
const modelValue = defineModel<string>();
```

I do not use Vue much, and this is untested and just from reading the docs: https://vuejs.org/api/sfc-script-setup.html#usage-with-typescript

File: stubs/inertia-vue/resources/js/Components/TextInput.vue [hunk 2]
```diff
@@ -1,14 +1,7 @@
...
 
-defineProps({
-    modelValue: {
-        type: String,
-        required: true,
-    },
-});
-
-defineEmits(['update:modelValue']);
+const modelValue = defineModel();
```
Comment (Jubeki) [inline stubs/inertia-vue/resources/js/Components/TextInput.vue:4]:
```suggestion
const modelValue = defineModel({
    type: String,
    required: true,
});
```
I do not use Vue much, and this is untested and just from reading the docs: https://vuejs.org/api/sfc-script-setup.html#definemodel

File: stubs/inertia-vue-ts/resources/js/Components/TextInput.vue (hunk 1 above)
Comment (Jubeki) [reply stubs/inertia-vue-ts/resources/js/Components/TextInput.vue:4]: Maybe add here the required part to? Seems inconsistent with the non typescript part ?
```suggestion
const modelValue = defineModel<string>({ required: true });
```

**Review by jessarcher — APPROVED** [review]

**Comment by jessarcher** [comment]
Thanks @osbre and @Jubeki!

**Pull request merged**